vignettes/lab_grading.Rmd

```diff
@@ -0,0 +1,395 @@
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup,
+    new_vars = vars(PARAMCD, PARAM, PARAMN),
+    by_vars = vars(LBTESTCD)
+  ) %>%
+  ## Calculate PARCAT1 AVAL AVALC ANRLO ANRHI
+  ## Dummy the values for BASE
+  mutate(
+    PARCAT1 = LBCAT,
+    AVAL = LBSTRESN,
+    AVALC = LBSTRESC,
+    ANRLO = LBSTNRLO,
+    ANRHI = LBSTNRHI,
+    BASE = AVAL - 10
+  )
+```
+
+Another look-up table is used to add on `ATOXDSCL` and `ATOXDSCH` using
+`PARAMCD`. `ATOXDSCL` holds the terms for grading low lab values, and `ATOXDSCH` holds 
+the terms for grading high lab values. The names of these variables can be user-defined.
+
+```{r, eval=TRUE}
+# Assign ATOXDSCL and ATOXDSCH to hold lab grading terms
+# ATOXDSCL and ATOXDSCH hold terms defined by NCI-CTCAEv4.
+grade_lookup <- tibble::tribble(
+  ~PARAMCD, ~ATOXDSCL,                      ~ATOXDSCH,
+  "ALB",     "Hypoalbuminemia",             NA_character_,
+  "ALKPH",   NA_character_,                 "Alkaline phosphatase increased",
+  "ALT",     NA_character_,                 "Alanine aminotransferase increased",
+  "AST",     NA_character_,                 "Aspartate aminotransferase increased",
+  "BILI",    NA_character_,                 "Blood bilirubin increased",
+  "CA",      "Hypocalcemia",                "Hypercalcemia",
+  "CHOLES",  NA_character_,                 "Cholesterol high",
+  "CK",      NA_character_,                 "CPK increased",
+  "CREAT",   NA_character_,                 "Creatinine increased",
+  "GGT",     NA_character_,                 "GGT increased",
+  "GLUC",    "Hypoglycemia",                "Hyperglycemia",
+  "HGB",     "Anemia",                      "Hemoglobin increased",
+  "POTAS",   "Hypokalemia",                 "Hyperkalemia",
+  "LYMPH",  "CD4 lymphocytes decreased",   NA_character_,
+  "PHOS",    "Hypophosphatemia",            NA_character_,
+  "PLAT",    "Platelet count decreased",    NA_character_,
+  "SODIUM",  "Hyponatremia",                "Hypernatremia",
+  "WBC",     "White blood cell decreased",  "Leukocytosis",
+)
+
+adlb <- adlb %>% 
+  derive_vars_merged(
+    dataset_add = grade_lookup,
+    by_vars = vars(PARAMCD),
+  )
+```
+
+It is now straightforward to create the grade, for low lab values the grade will
+be held in `ATOXGRL` and for high lab values the grade will be held in `ATOXGRH`
+
+```{r, eval=TRUE}
+adlb <- adlb %>% 
+  derive_var_atoxgr_dir(
+    new_var = ATOXGRL,
+    tox_description_var = ATOXDSCL,
+    criteria_direction = "L",
+    get_unit_expr = extract_unit(PARAM)
+  ) %>% 
+  derive_var_atoxgr_dir(
+    new_var = ATOXGRH,
+    tox_description_var = ATOXDSCH,
+    criteria_direction = "H",
+    get_unit_expr = extract_unit(PARAM)
+  )
+```
+
+Note: {admiral} does not grade 'Anemia' or 'Hemoglobin Increased' because the metadata is 
+based on the SI unit of 'g/L', however the CDISC data has SI unit of 'mmol/L'.
+Please see `SI_UNIT_CHECK` variable in {admiral} metadata `atoxgr_criteria_ctcv4`.
```

Comment:
```suggestion
Note: `{admiral}` does not grade 'Anemia' or 'Hemoglobin Increased' because the metadata is 
based on the SI unit of 'g/L', however the CDISC data has SI unit of 'mmol/L'.
Please see `SI_UNIT_CHECK` variable in `{admiral}` metadata `atoxgr_criteria_ctcv4`.
```

Reply:
done